packages/flame/lib/src/components/core/component.dart

```diff
@@ -927,13 +926,34 @@ class Component {
     }
   }
 
+  /// Used by the [FlameGame] to set the loaded state of the component, since
+  /// the game isn't going through the whole normal component life cycle.
   @internal
-  void setMounted() {
+  void setLoaded() {
     _setLoadedBit();
+    _loadCompleter?.complete();
+    _loadCompleter = null;
+  }
+
+  /// Used by the [FlameGame] to set the mounted state of the component, since
+  /// the game isn't going through the whole normal component life cycle.
+  @internal
+  void setMounted() {
```

Comment:
This method already existed, so I wonder if the docs here is correct? Other components that are not a flame game don't call it?

Reply:
It was previously only called by `FlameGame` too, but the completer wasn't completed here previously